packages/stripe_platform_interface/lib/src/method_channel_stripe.dart

```diff
@@ -127,21 +122,15 @@ class MethodChannelStripe extends StripePlatform {
   @override
   Future<PaymentIntent> handleCardAction(
       String paymentIntentClientSecret) async {
-    try {
-      final result = await _methodChannel
-          .invokeMapMethod<String, dynamic>('handleCardAction', {
-        'paymentIntentClientSecret': paymentIntentClientSecret,
-      });
-
-      final tmp = result?['paymentIntent'] as Map<String, dynamic>;
-
-      return PaymentIntent.fromJson(tmp.unfoldToNonNull());
-    } on Exception catch (_) {
-      throw const StripeError<PaymentIntentError>(
-        code: PaymentIntentError.unknown,
-        message: 'Handle  payment intent for card failed',
-      );
-    }
+    final result = await _methodChannel
+        .invokeMapMethod<String, dynamic>('handleCardAction', {
+      'paymentIntentClientSecret': paymentIntentClientSecret,
+    });
+
+    final tmp = ResultParser<PaymentIntent>(
+            parseJson: (json) => PaymentIntent.fromJson(json))
+        .parse(result: result!, successResultKey: 'paymentIntent');
+    return tmp;
```

Comment:
let's get rid of all 'tmp' variables throughout the file, also result shouldn't be forced to be not null